playwright-tests/tests/editor.spec.js

```diff
@@ -276,4 +276,84 @@ test.describe("?page=projects&tab=editor", () => {
       await expect(warningText).toBeVisible();
     });
   });
+
+  test.describe("Import project from NEAR Catalog", () => {
+    test.beforeEach(async ({ page }) => {
+      await page.goto(`/${ROOT_SRC}?page=projects&tab=catalogImport`);
+    });
+    test.use({
+      storageState: "playwright-tests/storage-states/wallet-connected.json",
+    });
+
+    // should have visible projects
+    test("should have visible projects", async ({ page }) => {
+      const projectHeading = page.getByRole("heading", {
+        name: "Race of Sloths",
+      });
+      await expect(projectHeading).toBeVisible({ timeout: 60000 });
+    });
+
+    test("Should be able to create a new project from NEAR Catalog", async ({
+      page,
+    }) => {
+      const projectHeading = page.getByRole("heading", {
+        name: "Race of Sloths",
+      });
+      await expect(projectHeading).toBeVisible({ timeout: 60000 });
+
+      const parentDiv = page
+        .locator("div.project-card")
+        .filter({ has: projectHeading });
+      await expect(parentDiv).toBeVisible({ timeout: 60000 });
+      await parentDiv.dispatchEvent("click");
+
+      const nextButton = page.getByRole("button", { name: "Next" });
+      await expect(nextButton).toBeVisible({ timeout: 60000 });
+      await nextButton.click();
+
+      const createButton = page.getByRole("button", { name: "Create" });
+      await expect(createButton).toBeVisible({ timeout: 60000 });
+      await createButton.click();
+
+      const expectedTransactionData = {
```

Comment:
What happens when the data changes? Now test breaks

What if we mock the fetch request? 

```
await page.route(
'**/*' // replace with regex for the request url
, async (route, request) => {
  // Override headers
  const headers = {
    ...request.headers(),
    foo: 'foo-value', // set "foo" header
    bar: undefined, // remove "bar" header
  };
  await route.continue({ headers });
});
```

Add "mocking a fetch request" to the test cookbook

I'm sure you can find a good example for yourself in here: 
https://github.com/NEAR-DevHub/neardevhub-bos/tree/main/playwright-tests

Reply:
And when we mock it, let's mock with much smaller project data

Reply:
"We already have a test that confirms the transactionObj -- we really just need this test to confirm that the fields are prepopulated"

Then isn't this not needed?

Reply:
I wouldn't say the transaction object check itself is needed, but should definitely validate pre-populated fields

After clicking a project card, assert that input values on Editor are equal to what was provided

Reply:
Would be doing just that right now, if it weren't for Timeout errors